### ISSUE TITLE
BatchNorm cannot be fused in some cases

### ISSUE THREAD

**Issue by veralauee** [onnx/tensorflow-onnx#2186]
**Describe the bug**
FusedBatchNormV3 cannot be fused into Conv when the Conv is the output of the previous BN. For example,
`Conv1 -> BN1 -> Conv2 -> BN2`
BN1 can be fused, but BN2 is kept. Ideally both should be fused into their Convs. 
I think the logic of  [has_dependencies](https://github.com/peri044/tensorflow-onnx/blob/12d3a8b634e0d5a0b38a701d3bc908f77786d691/tf2onnx/optimizer/back_to_back_optimizer.py#L45-L50) is not accurate.

TF model looks like
![image](https://github.com/onnx/tensorflow-onnx/assets/39462115/1622e503-a880-4511-a7c2-f74289deb078)

Converted ONNX looks like
![image](https://github.com/onnx/tensorflow-onnx/assets/39462115/3ff9ec5c-ea44-417b-a683-1ba537a25182)

**Urgency**
It would be better to fix it, otherwise the latency could be slower than expected. 

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 18.04*): Linux Ubuntu 20.04
- TensorFlow Version: 2.12.0
- Python version: 3.8
- ONNX version (if applicable, e.g. 1.11*): 1.14.0
- ONNXRuntime version (if applicable, e.g. 1.11*):
